src/cmd/csi_driver.go

```diff
@@ -0,0 +1,99 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+//var (
+//	nodeID    string
+//	endpoint  string
+//	probeAddr string
+//)
+//
+//func csiDriverFlags() *pflag.FlagSet {
+//	csiDriverFlags := pflag.NewFlagSet("csi-driver", pflag.ExitOnError)
+//	csiDriverFlags.StringVar(&nodeID, "node-id", "", "node id")
+//	csiDriverFlags.StringVar(&endpoint, "endpoint", "unix:///tmp/csi.sock", "CSI endpoint")
+//	csiDriverFlags.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+//	return csiDriverFlags
+//}
+//
+//func setupCSIDriver(ns string, cfg *rest.Config) (manager.Manager, func(), error) {
+//	defaultUmask := unix.Umask(0000)
+//	cleanUp := func() {
+//		unix.Umask(defaultUmask)
+//	}
+//
+//	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+//		Namespace:              ns,
+//		Scheme:                 scheme.Scheme,
+//		MetricsBindAddress:     ":8080",
+//		Port:                   8383,
+//		HealthProbeBindAddress: probeAddr,
+//		LivenessEndpointName:   "/livez",
+//	})
+//	if err != nil {
+//		log.Error(err, "unable to start manager")
+//		return nil, cleanUp, err
+//	}
+//
+//	csiOpts := dtcsi.CSIOptions{
+//		NodeID:   nodeID,
+//		Endpoint: endpoint,
+//		RootDir:  dtcsi.DataPath,
+//	}
+//
+//	fs := afero.NewOsFs()
+//
+//	if err := fs.MkdirAll(filepath.Join(csiOpts.RootDir), 0770); err != nil {
+//		log.Error(err, "unable to create data directory for CSI Driver")
+//		return nil, cleanUp, err
+//	}
+//
+//	access, err := metadata.NewAccess(dtcsi.MetadataAccessPath)
+//	if err != nil {
+//		log.Error(err, "failed to setup database storage for CSI Driver")
+//		os.Exit(1)
+//	}
+//	if err := metadata.CorrectMetadata(mgr.GetClient(), access); err != nil {
+//		log.Error(err, "failed to correct database storage for CSI Driver")
+//	}
+//
+//	if err := csidriver.NewServer(mgr.GetClient(), csiOpts, access).SetupWithManager(mgr); err != nil {
+//		log.Error(err, "unable to create CSI Driver server")
+//		return nil, cleanUp, err
+//	}
+//
+//	if err := csiprovisioner.NewOneAgentProvisioner(mgr, csiOpts, access).SetupWithManager(mgr); err != nil {
+//		log.Error(err, "unable to create CSI Provisioner")
+//		return nil, cleanUp, err
+//	}
+//
+//	if err := mgr.AddHealthzCheck("livez", healthz.Ping); err != nil {
+//		log.Error(err, "unable to set up health check")
+//		return nil, cleanUp, err
+//	}
+//
+//	if err := csigc.NewCSIGarbageCollector(mgr.GetClient(), csiOpts, access).SetupWithManager(mgr); err != nil {
+//		log.Error(err, "unable to create CSI Garbage Collector")
+//		return nil, cleanUp, err
+//	}
+//
+//	return mgr, cleanUp, nil
+//}
```

Comment:
Leftovers

Reply:
kept for reference, should've marked it as a draft. Leftovers are deleted now